### PR TITLE
add missing hatchling dependency

### LINKED ISSUE #101: Elasticsearch 8.15.0 not available
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Following conda-forge's [FAQ](https://conda-forge.org/docs/user/faq/#the-feedstock-for-a-package-from-conda-forge-is-updated-how-long-should-it-take-to-update-on-anaconda-cloud), I'm opening this issue because it has passed 24h since the 8.15.0 [PR](https://github.com/conda-forge/elasticsearch-feedstock/pull/99) was merged and the package is still not available on conda-forge.
It looks like the main branch [build job](https://github.com/conda-forge/elasticsearch-feedstock/runs/28996115195) failed with `ModuleNotFoundError: No module named 'hatchling'`. Should it be a `host` dependency in the recipe?

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

### PR DESCRIPTION
**Pull request by miguelgrinberg**
Fixes #101

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by miguelgrinberg** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,18 +13,19 @@ source:
...
 build:
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install . --no-deps -vv
   noarch: python
 
 requirements:
   host:
-    - python >=3.6
+    - python >=3.7
```
Comment (pquentin) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.8
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,18 +13,19 @@ source:
...
+    - hatchling
     - pip
     - elastic-transport >=8.13,<9
 
   run:
-    - python >=3.6
+    - python >=3.7
```
Comment (pquentin) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.8
```

**Review by pquentin — APPROVED** [review]
Thanks! LGTM. Note that as a follow-up we could add optional dependencies too: https://github.com/conda-forge/urllib3-feedstock/blob/53a3fd4c3bcfccc6dd554493ad0ca6ef280a0b08/recipe/meta.yaml#L25-L32

**Pull request merged**